### PR TITLE
remove_dynamic_fee -> Done

### PR DESCRIPTION
**Pull request by vutt94**

### CODE REVIEW

File: test/dmmPoolV2.js [hunk 1]
```diff
@@ -0,0 +1,802 @@
...
+let pool;
+let trader;
+let feeTo;
+let liquidityProvider;
+let app;
+
+let ampBps = new BN(20000);
+let unamplifiedBps = new BN(10000);
+
+let baseRate = new BN(0);
```
Comment (Zohar97) [inline test/dmmPoolV2.js]: remove unused variable

Comment (vutt94) [reply test/dmmPoolV2.js]: It's gone

**Review by Zohar97 — APPROVED** [review]

File: contracts/DMMPoolV2.sol [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+        uint256 amount0In,
+        uint256 amount1In,
+        uint256 amount0Out,
+        uint256 amount1Out,
+        address indexed to,
+        uint256 feeInPrecision
+    );
+    event Sync(uint256 vReserve0, uint256 vReserve1, uint256 reserve0, uint256 reserve1);
+
+    constructor() public ERC20Permit("KyberDMM LP", "DMM-LP", "1") {
```
Comment (manhlx3006) [inline contracts/DMMPoolV2.sol]: Please change to "KyberSwap LP", "KS-LP" instead
Should change all DMM -> KS, KyberDMM -> KyberSwap

File: contracts/DMMPoolV2.sol [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+        _reserve0 = reserve0;
+        _reserve1 = reserve1;
+        uint32 _ampBps = ampBps;
+        _vReserve0 = vReserve0;
+        _vReserve1 = vReserve1;
+        if (_ampBps == BPS) {
+            _vReserve0 = _reserve0;
+            _vReserve1 = _reserve1;
+        }
+        feeInPrecision = getFinalFee(FeeFomula.getFee(factorInPrecision), _ampBps);
```
Comment (manhlx3006) [inline contracts/DMMPoolV2.sol]: why do we use FeeFomula.getFee(factorInPrecision) for feeInPrecision?
It should be fixed when Factory initializes the pool contract

**Review by manhlx3006 — CHANGES_REQUESTED** [review]

File: contracts/KSFactory.sol [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            getUnamplifiedPool[token0][token1] = pool;
+            getUnamplifiedPool[token1][token0] = pool;
+        }
+        allPools.push(pool);
+
+        emit PoolCreated(token0, token1, pool, ampBps, feeBps, allPools.length);
+    }
+
+    function setFeeConfiguration(address _feeTo, uint16 _governmentFeeBps) external override {
+        require(msg.sender == feeToSetter, "KS: FORBIDDEN");
```
Comment (Zohar97) [inline contracts/KSFactory.sol]: require(msg.sender == feeToSetter, "KS: FORBIDDEN");
Should use modifier. Also with enableFeeOption, disableFeeOption, setFeeToSetter

File: contracts/DMMPoolV2.sol (hunk 2 above)
Comment (vutt94) [reply contracts/DMMPoolV2.sol]: Done

File: contracts/DMMPoolV2.sol (hunk 3 above)
Comment (vutt94) [reply contracts/DMMPoolV2.sol]: Fixed. feeInPrecision is now fixed.

File: contracts/KSFactory.sol (hunk 4 above)
Comment (vutt94) [reply contracts/KSFactory.sol]: Added modifier to resolve this

File: contracts/KSFactory.sol [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    uint256 internal constant BPS = 10000;
+
+    address private feeTo;
+    uint16 private governmentFeeBps;
+    address public override feeToSetter;
+
+    /// @dev fee to set for pools
+    mapping(uint16 => bool) private feeOptions;
```
Comment (manhlx3006) [inline contracts/KSFactory.sol]: how to check if a fee option has been enabled if it is private?

File: contracts/KSPool.sol [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+        IERC20 _token1,
+        uint32 _ampBps,
+        uint16 _feeBps
+    ) external {
+        require(msg.sender == address(factory), "KS: FORBIDDEN");
+        token0 = _token0;
+        token1 = _token1;
+        ampBps = _ampBps;
+        uint256 finalFee = (_feeBps * PRECISION) / BPS;
+        feeInPrecision = getFinalFee(finalFee, _ampBps);
```
Comment (manhlx3006) [inline contracts/KSPool.sol]: there is no amp applying here as we already pass the fee value

**Review by manhlx3006 — CHANGES_REQUESTED** [review]

File: contracts/KSFactory.sol (hunk 5 above)
Comment (vutt94) [reply contracts/KSFactory.sol]: It's is public now.

File: contracts/KSPool.sol (hunk 6 above)
Comment (vutt94) [reply contracts/KSPool.sol]: Removed getFinalFee.

**Review by Zohar97 — APPROVED** [review]
LGTM!!!

File: contracts/KSPool.sol [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+        ReserveData memory newData;
+        {
+            // scope for _token{0,1}, avoids stack too deep errors
+            IERC20 _token0 = token0;
+            IERC20 _token1 = token1;
+            require(to != address(_token0) && to != address(_token1), "KS: INVALID_TO");
+            if (amount0Out > 0) _token0.safeTransfer(to, amount0Out); // optimistically transfer tokens
+            if (amount1Out > 0) _token1.safeTransfer(to, amount1Out); // optimistically transfer tokens
+            if (callbackData.length > 0)
+                IKSCallee(to).dmmSwapCall(msg.sender, amount0Out, amount1Out, callbackData);
```
Comment (manhlx3006) [inline contracts/KSPool.sol]: can you also change this to "kyberSwapCall"?

Comment (vutt94) [reply contracts/KSPool.sol]: Done

**Review by manhlx3006 — APPROVED** [review]

**Review by Zohar97 — APPROVED** [review]
LGTM!!!

**Pull request merged**